### ISSUE TITLE
"N snapshots changed their representation" but update doesn't do anything

### ISSUE THREAD

**Issue by pawamoy** [15r10nk/inline-snapshot#79]
I get this pytest output:

```
======================== inline snapshot ========================
Info: 1 snapshots changed their representation (--inline-snapshot=update)
```

With a clean repository state, I run pytest again with `--inline-snapshot=update`:

```
======================== inline snapshot ========================
updated 1 snapshots
```

However there are no files changed.

I'll run a debug session to see what happens.

**Comment by pawamoy** [comment]
For some reason, `self._token_of_node(old_node) != new_token` is True for the same code. In the following screenshot, you can see that `self._token_of_node(old_node)` yield `(3, "'3603a08e7614*.html'")` while in `new_token` the same element is `simple_token(type=3, string='"3603a08e7614*.html"')`.

![codetokens](https://github.com/15r10nk/inline-snapshot/assets/3999221/b1c2ca14-8f89-4627-b0b6-0bdcb143c365)

**Comment by pawamoy** [comment]
Looks like `normalize_strings` sometimes yields tuples without re-wrapping them in `simple_token`.

**Comment by pawamoy** [comment]
Ah, in one case the string token value is `'3603a08e7614*.html'` and in the other it's `"3603a08e7614*.html"` (single quotes vs double quotes). I suppose the new token should normalize strings too.

**Comment by pawamoy** [comment]
With these two changes it fixes the issue above:

```python
# in EqValue._get_changes.<locals>.check
# generic fallback
new_token = list(normalize_strings(value_to_token(new_value)))
```

```python
# in normalize_strings, always yield simple_token instances
...
        if current_string is not None:
            yield simple_token(token.STRING, repr(current_string))
            current_string = None

        yield t

    if current_string is not None:
        yield simple_token(token.STRING, repr(current_string))
```

**Comment by pawamoy** [comment]
Hmmm stil getting the message though, investigating.

**Comment by pawamoy** [comment]
Right, this time this might indeed be caused by explicit tuples, or trailing commas.

Up to element 36, both new token and old token are equal. Then old token has an additional comma before closing with `)`. The rest resumes (is equal) but is shifted by one. More shifts happen each time there's trailing comma.

![tokens](https://github.com/15r10nk/inline-snapshot/assets/3999221/049a9584-613b-4de6-887c-04bb9213349b)

**Comment by 15r10nk** [comment]
Thank you for the debugging. I will look into this tomorrow.

**Comment by 15r10nk** [comment]
#82 which is part of 0.9.0 fixes the issue with the string quoting.
